File: greenDb/src/main/java/com/xy/db/base/StringHashMapConverter.java
package com.xy.db.base;


import org.greenrobot.greendao.converter.PropertyConverter;
import org.json.JSONException;
import org.json.JSONObject;
import java.util.HashMap;
import java.util.Iterator;

public class StringHashMapConverter implements PropertyConverter<HashMap<String,String>, String> {

    @Override
    public HashMap<String,String> convertToEntityProperty(String databaseValue) {
        if (databaseValue == null) {
            return null;
        } else {
            HashMap<String,String> hashMap = new HashMap<>();
            try {
                JSONObject jsonObject = new JSONObject(databaseValue);
                Iterator it = jsonObject.keys();
                while(it.hasNext()){
                    String key = it.next().toString();
                    String value = (String)jsonObject.get(key).toString();
                    hashMap.put(key, value);
                }
            } catch (JSONException e) {
                e.printStackTrace();
            }
            return hashMap;
        }
    }

    @Override
    public String convertToDatabaseValue(HashMap<String, String> hashMap) {
        JSONObject jsonObject = new JSONObject();
        if (hashMap != null) {
            for (String key : hashMap.keySet()) {
                try {
                    jsonObject.put(key, hashMap.get(key));
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }
        return jsonObject.toString();
    }
}

File: LibBase/src/main/java/com/xy/baselib/db/StringHashMapConverter.kt
package com.xy.baselib.db

import org.greenrobot.greendao.converter.PropertyConverter
import org.json.JSONException
import org.json.JSONObject

class StringHashMapConverter : PropertyConverter<HashMap<String, String>?, String?> {
    override fun convertToEntityProperty(databaseValue: String?): HashMap<String, String>? {
        return if (databaseValue == null) {
            null
        } else {
            val hashMap = HashMap<String, String>()
            try {
                val jsonObject = JSONObject(databaseValue)
                val it: Iterator<*> = jsonObject.keys()
                while (it.hasNext()) {
                    val key = it.next().toString()
                    hashMap[key] = jsonObject[key].toString()
                }
            } catch (e: JSONException) {
                e.printStackTrace()
            }
            hashMap
        }
    }

    override fun convertToDatabaseValue(hashMap: HashMap<String, String>?): String? {
        val jsonObject = JSONObject()
        if (hashMap != null) {
            for (key in hashMap.keys) {
                try {
                    jsonObject.put(key, hashMap[key])
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
        }
        return jsonObject.toString()
    }
}